sympy/stats/stochastic_process_types.py

```diff
@@ -225,7 +241,8 @@ def __call__(self, time):
         if time not in self.index_set:
             raise IndexError("%s is not in the index set of %s"%(time, self.symbol))
         func_obj = Function(self.symbol)(time)
-        pspace_obj = StochasticPSpace(self.symbol, self)
+        distribution = getattr(self, 'distribution', None)
```

Comment:
But, `self.distribution` will lead to AttributeError when the process does not have distribution. So, to avoid try-except blocks I have used `getattr`

Reply:
I would suggest to add a property `distribution` in `StochasticProcess` which can return `None`.

Reply:
But when distribution takes an argument apart from `self` as here:
```python
def distribution(self, rv):
        return PoissonDistribution(self.lamda*rv.key)
```

Reply:
> But, `self.distribution` will lead to AttributeError when the process does not have distribution. So, to avoid try-except blocks I have used `getattr`

Python allows a lot of freedom to manipulate attributes of objects, but it's usually not recommended to use all of this freedom.

Try to code as if you were writing C++ code. So, either `self` has `.distribution` as attribute, or it does not. If only some subclasses of this class have `.distribution`, then this method should be lowered to a subclass.

Reply:
> I would suggest to add a property distribution in StochasticProcess which can return None.

Even better would be an abstract method.